hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/mapred/TestCopyMapper.java

```diff
@@ -444,6 +449,54 @@ private void testCopyingExistingFiles(FileSystem fs, CopyMapper copyMapper,
     }
   }
 
+  @Test
+  public void testCopyWhileAppend() throws Exception {
+    deleteState();
+    mkdirs(SOURCE_PATH + "/1");
+    touchFile(SOURCE_PATH + "/1/3");
+    CopyMapper copyMapper = new CopyMapper();
+    StubContext stubContext = new StubContext(getConfiguration(), null, 0);
+    Mapper<Text, CopyListingFileStatus, Text, Text>.Context context =
+            stubContext.getContext();
+    copyMapper.setup(context);
+    final Path path = new Path(SOURCE_PATH + "/1/3");
+    int manyBytes = 100000000;
+    appendFile(path, manyBytes);
+    ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+    Runnable task = new Runnable() {
+      public void run() {
+        try {
+          int maxAppendAttempts = 20;
+          int appendCount = 0;
+          while (appendCount < maxAppendAttempts) {
+            appendFile(path, 1000);
+            Thread.sleep(200);
+            appendCount++;
+          }
+        } catch (IOException | InterruptedException e) {
+          e.printStackTrace();
+        }
+      }
+    };
+    scheduledExecutorService.schedule(task, 10, TimeUnit.MILLISECONDS);
+    boolean isFileMismatchErrorPresent = false;
+    try {
```

Comment:
I think you can use LambdaTestUtils.intercept for this since all you are doing is asserting that an operation raises an IOException; catch that and check its contents, rethrowing if it doesn't contain the specific text constants defined in the production code.

Reply:
I may be wrong but as per my understanding of intercept() method it should be used when we are expecting some Exception but in our scenario what we want is copy should run successfully without any exception. If there is any exception we should fail the test.

Reply:
good point. But: why then do you catch some exceptions and discard them? What other IOEs get raised which aren't errors?

Reply:
There can be other exceptions while running a distcp like CopyReadException when we failed to read the data from the source file. 
The main idea of this test case is to validate that there is no copy failure with specific reasons like length/checksum mismatch while we append a file when copy is going on. This is the reason of catching specific exceptions. 
I believe other exceptions have already been tested as a part of other tests.

Reply:
OK